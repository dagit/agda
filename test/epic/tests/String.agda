module tests.String where

open import Prelude.IO
open import Prelude.List
open import Prelude.String
open import Prelude.Unit


testString : String
testString = "To boldly go where no man gone before"

printList : forall {A} -> (A -> IO Unit) -> List A -> IO Unit
printList p [] = return unit
printList p (x :: xs) =
    p x ,,
    printList p xs

main : IO Unit
main =
    putStrLn testString ,,
    printList printChar (fromString testString) ,,
    putStrLn "" ,,
    putStrLn (fromList (fromString testString)) ,,
    return unit
