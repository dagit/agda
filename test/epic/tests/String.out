To boldly go where no man gone before
To boldly go where no man gone before
To boldly go where no man gone before
